:seedling: Decompose RegistryV1ToHelmChart function

perdasilva opened the pull request:
# Description

Refactor `RegistryV1ToHelmChart` function in the `rukpak/convert` package by taking out the code that parses a bundle filesystem into a `RegistryV1` object and putting it into its own function `ParseFS`. This facilitates manipulation of bundles in other activities, e.g. tooling/CLI

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

tmshort commented:
broken unit test due to removal of function.

camilamacedo86 reviewed (commented):
Some lint issues otherwise it seems LGTM to me 👍

perdasilva commented:
> broken unit test due to removal of function.

Sorry about that. Shot from the hip as I was getting rushed off to dinner. I've fixed it up by removing the unit test. The test below it performs the same test. I also moved it to the test package to help us avoid testing private functions in the future.

perdasilva commented on internal/rukpak/convert/registryv1_test.go:
```diff
@@ -438,50 +445,68 @@ func TestRegistryV1SuiteGenerateErrorAllNamespaceDisabled(t *testing.T) {
 	require.Nil(t, plainBundle)
 }
 
-func TestRegistryV1SuiteGeneratePropagateCsvAnnotations(t *testing.T) {
```
Removed `TestRegistryV1SuiteGeneratePropagateCsvAnnotations` since it was testing an unexported function and the behavior is also tested on line 489

camilamacedo86 commented on internal/rukpak/convert/registryv1_test.go:
```diff
@@ -559,10 +584,47 @@ func convertToUnstructured(t *testing.T, obj interface{}) unstructured.Unstructu
+spec:
+  installModes:
+    - type: AllNamespaces
+      supported: true
```
I can see here that we still using the same mock from what was removed `csv.Spec.InstallModes = []v1alpha1.InstallMode{{Type: v1alpha1.InstallModeTypeMultiNamespace, Supported: true}}`

camilamacedo86 commented on internal/rukpak/convert/registryv1.go:
```diff
@@ -42,20 +42,57 @@ type Plain struct {
+			Name: fmt.Sprintf("object-%x.json", hash[0:8]),
+			Data: jsonData,
+		})
+	}
```
👍 I can see that we are doing mainly the same just/refac/cleanup

camilamacedo86 reviewed (approved):
/lgtm

LalatenduMohanty commented:
@perdasilva #1737 got merged, so this PR need a rebase.

camilamacedo86 reviewed (approved):
/lgtm

When pass in the CI ( flake )
